app.py

```diff
@@ -52,7 +54,8 @@ def logout():
 
 @app.route("/graphcall")
 def graphcall():
-    token = _get_token_from_cache(app_config.SCOPE)
+    token = azureAuth.get_token_from_cache(session)
+    session["token_cache"] = azureAuth.save_cache()
```

Comment:
This line is a good catch! We should have had it in our original sample, even without this refactor PR. If you can start a separated PR containing this fix only, we would be happily take it (and then your contribution would be permanently recorded in this code base)!

Reply:
Thank you. You're referring to the `session["token_cache"] = azureAuth.save_cache()` step correct? I believe the original code still functions the same where `_get_token_from_cache(app_config.SCOPE)`  will call `_save_cache(cache)` if the cached account belongs to the current signed in user. This save cache function then performs the same operation I have done in the line you called in this thread `session["token_cache"] = cache.serialize()`. Is my logic in the lines you've called out here different then this?

Reply:
Oh you are right. I was assuming your refactored code would have exactly the same behaviors as previous, so that your adding this line might mean I forgot that in the original code. Now looking closer, I can see you just move that line from inside the `get_token_from_cache(session)` to here.

OK then. Both the before and after this PR code would work. Again, we will take all these into consideration when we refactor this sample next time. (And we will!)